### DISCUSSION TITLE
Virtual control panel working! How do I arm/disarm from automation?

### DISCUSSION THREAD

**Discussion by yakirg** [Dilbert66/esphome-vistaECP#129] [General]
I got the system working (esp8266 + vista 20p).
It took a min, but got it working to the point I can arm/disarm from the virtual keypad in HA.
I am looking for the documented services:
HA Services
Basic alarm services. These services default to partition 1:

"alarm_disarm", Parameter: "code" (access code)
"alarm_arm_home"
"alarm_arm_night", Parameter: "code" (access code)
"alarm_arm_away"
"alarm_trigger_panic"
"alarm_trigger_fire"

I can't find where these are defined, or how I cause the automation to trigger them. (I am looking to have the alarm arm at night, and disarm in the morning).

Any idea what I'm missing?

**Comment by yakirg** [comment]
I think I found it!
Adding an action from ESPHome: alarmgateway_alarm_disarm or ESPHome: alarmgateway_alarm_arm_home
Seems to work!!

**Comment by Dilbert66** [comment]
Yes, that's it.  You can easily identify services provided by searching in the Developer Tools section under the Services tab.

**Comment by yakirg** [comment]
Thank you SO much for this service, and the support!!!
